File: C++/M01_ProgrammingBasics/L05_WhileLoop/Exercises/Solutions/P01_OldBooks.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {

	string book;
	getline(cin, book);
	int capacity;
	cin >> capacity;

	string searcher;
	getline(cin, searcher);
	int counter = 0;
	while (searcher != book) {

		if (counter == capacity) {
			cout << "The book you search is not here!" << endl;
			cout << "You checked " << capacity << " books." << endl;

			return 0;
		}
		getline(cin, searcher);
		counter++;
	}

	if (searcher == book && counter != 0) {
		cout << "You checked " << counter - 1 << " books and found it." << endl;
	} else {
		cout << "You checked " << capacity << " books and found it." << endl;
	}

	return 0;
}

File: Python/M01_ProgrammingBasics/L05_WhileLoop/Exercises/Solutions/P01_OldBooks.py
needed_book = input()
counter = 0

while True:
    line = input()

    if line == needed_book:
        print(f"You checked {counter} books and found it.")
        break

    if line == "No More Books":
        print(f"The book you search is not here!")
        print(f"You checked {counter} books.")
        break

    counter += 1